### DISCUSSION TITLE
possible to get this in wasm for use in other programming languages?

### DISCUSSION THREAD

**Discussion by csells** [ohmjs/ohm#348] [Ideas]
I want to use this from Dart and Rust, too, not just JS. It's too awesome for just one language!

**Comment by pdubroy** [comment]
@csells Thanks :-)

I'd be interested in supporting Ohm in other languages. I can see how wasm would work for parsing, but I'd imagine that it would be difficult to do semantic actions that way, since it involves many calls out into the host language. What do you think?

A while back I thought about trying to use [AssemblyScript](https://github.com/AssemblyScript/assemblyscript) to produce a wasm version of Ohm. I would be cool to try, if you or anyone else is up for experimenting with it :-)

**Comment by mlajtos** [comment]
Having just an Ohm parser generating CST inside Swift (or any other language) would be awesome. Ohm/Swift would be dream come true.

(This got me thinking. There is quite straightforward way to use [JS inside Swift via JavaScriptCore](https://stackoverflow.com/a/37435316). I will look into wrapping Ohm/JS via this mechanism into Swift. Shouldn't take too much time to get something working. Edit: I am using Ohm via JavaScriptCore to generate AST for my lang and then AST is processed via JSON Decoder into native Swift data structure. It just works. 😅)

**Comment by tanishqkancharla** [comment]
I don't mean to advertise a different tool, but if specifically for Rust, Pest is fairly similar (i haven't tried myself, just seen it before): [https://pest.rs](https://pest.rs)
